crates/ruff_linter/src/rules/flake8_pyi/rules/non_self_return_type.rs

```diff
@@ -93,10 +97,67 @@ impl Violation for NonSelfReturnType {
     }
 
     fn fix_title(&self) -> Option<String> {
-        Some("Consider using `typing_extensions.Self` as return type".to_string())
+        Some("Use `Self` as return type".to_string())
     }
 }
 
+fn import_self_edit(checker: &Checker, range: TextRange) -> Option<Edit> {
+    let target_version = checker.settings.target_version.as_tuple();
+    let source_module = if checker.source_type.is_stub() || target_version < (3, 11) {
+        "typing_extensions"
+    } else {
+        "typing"
```

Comment:
```suggestion
    let source_module = if checker.source_type.is_stub() || target_version >= (3, 11) {
        "typing"
    } else {
        "typing_extensions"
```

Reply:
I think we _do_ want to use `typing` in stubs.

Reply:
@InSyncWithFoo is correct -- `Self` cannot be imported from `typing` in a `.py` file or a `.pyi` file if you want the code to support Python 3.10 or lower (see also: https://github.com/astral-sh/ruff/issues/9761).

In typeshed we always import `Self` from `typing_extensions`, because typeshed pretends that `typing_extensions` is part of the stdlib, and we want our stubs to support Python 3.8-3.13.

Reply:
Thanks both! I assumed the semantics were similar to PEP 585.